servicetalk-concurrent-api/docs/modules/ROOT/pages/blocking-safe-by-default.adoc

```diff
@@ -42,121 +43,27 @@ The implication of the above approach is the following:
-request being received and user code making sure that the request is not sent before an item is emitted.
-
-In order to avoid such scenarios, it is handy to follow certain best practices while writing blocking code in operators:
-
-- Avoid co-ordination between two operators on the same source.
-- If such co-ordination is required, try limiting such coordination in either data or control path but not
-inter-dependent on each other.
-- If co-ordination is required between data and control path, be aware of ReactiveStreams semantics and how the two
-paths interact with each other.
-
-If these rules are followed the above example can be modified to:
-
-[source, java]
-----
-    CountDownLatch latch = new CountDownLatch(1);
-    Publisher.from(1, 2, 3, 4)
-            .afterOnNext(integer -> {
-                latch.countDown();
-            })
-            .afterRequest(requestN -> { # <1>
-                latch.await();
-            });
-----
-<1> Use `afterRequest` which happens **after** `requestN` is delivered to the source.
-
-In this modified example, since we now use `afterRequest`, instead of `beforeRequest`, we do not block `requestN` to
-go to the source and this code is safe.
+== Task Offloading
+ServiceTalk uses Netty for network I/O and the Netty implementation uses a fixed number of I/O threads for executing
+network operations. To maximize throughput and minimize latency it is important to ensure that the I/O threads are
+always available to respond immediately to network activity. One approach to ensure I/O thread availability is to
+carefully limit the scope of work done by I/O threads and, whenever practical, delegate all other necessary tasks that
+are not related to I/O to some other thread. Moving tasks from I/O threads to other threads is called “offloading” and
+is a core technique used by ServiceTalk.
+
+ServiceTalk will, by default, execute most application code on threads other than the Netty I/O threads. There are some
+situations where offloading may not be applied; where APIs are synchronous, assumed unlikely to block, and, therefore,
+not specifying offloading characteristics simplifies the ServiceTalk APIs. For example, the `Predicate` provided to the `appendServiceFilter(Predicate<StreamingHttpRequest>,
+StreamingHttpServiceFilterFactory)` method of link:{source-root}/servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpServerBuilder.java[HttpServerBuilder] by an application must not block and may be executed on an I/O thread.
+
+For most invocations of application code, if the application developer knows that their code cannot block and always
+executes quickly in near constant time they can request that ServiceTalk not offload their code. This will improve
+application performance by reducing latency and overhead. Requests to not offload will be honored by ServiceTalk if all
+the other components in the same execution path have also opted out of offloading. As a last resort, tasks may also
```

Comment:
`Requests to not offload will be honored by ServiceTalk if all
 the other components in the same execution path have also opted out of offloading.` -> we haven't talked about opt-ing out of offloading at this point, can we link to that section of the doc so folks can get more context?